contracts/voting-verifier/src/query.rs

```diff
@@ -0,0 +1,26 @@
+use cosmwasm_std::{Deps, StdResult};
+
+use connection_router::state::Message;
+
+use crate::error::ContractError;
+use crate::state::VERIFIED_MESSAGES;
+
+pub fn verification_statuses(deps: Deps, messages: Vec<Message>) -> StdResult<Vec<(String, bool)>> {
+    messages
+        .into_iter()
+        .map(|message| {
+            is_message_verified(deps, &message).map(|verified| (message.id.to_string(), verified))
+        })
+        .collect::<Result<Vec<(_, _)>, _>>()
+        .map_err(Into::into)
+}
+
+pub fn is_message_verified(deps: Deps, message: &Message) -> Result<bool, ContractError> {
+    match VERIFIED_MESSAGES.may_load(deps.storage, &message.id)? {
+        Some(stored) if stored != *message => {
+            Err(ContractError::MessageMismatch(message.id.to_string()))
+        }
+        Some(_) => Ok(true),
+        None => Ok(false),
+    }
+}
```

Comment:
If I understand correctly how this will work, the whole call will fail if a single msg ID is false. I'd rather expect an unknown msg ID to give me `false`. To that end, I don't think either of these functions should have `Result` as return value

Reply:
The whole call will fail if the caller provides a different message with an already verified id, means the message batch is invalid.

An unknown msg id will return false.